File: internal/adapters/http/dto/installment.go
package dto

type CreateInstallmentRequest struct {
	Description       string  `json:"description"`
	AmountMode        string  `json:"amount_mode"`
	TotalAmount       float64 `json:"total_amount"`
	InstallmentAmount float64 `json:"installment_amount"`
	Count             int32   `json:"count"`
	CategoryID        int32   `json:"category_id"`
	PaymentMethodID   int32   `json:"payment_method_id"`
	PurchaseDate      string  `json:"purchase_date"` // YYYY-MM-DD
}

type InstallmentPlanResponse struct {
	ID                int32   `json:"id"`
	Description       string  `json:"description"`
	TotalAmount       float64 `json:"total_amount"`
	InstallmentCount  int32   `json:"installment_count"`
	InstallmentAmount float64 `json:"installment_amount"`
	StartMonth        string  `json:"start_month"`
	PaymentMethodID   int32   `json:"payment_method_id"`
}

File: frontend/layers/payment-methods/types/installment.ts
export type InstallmentAmountMode = 'TOTAL' | 'INSTALLMENT'

export interface CreateInstallmentRequest {
  description: string
  amount_mode?: InstallmentAmountMode
  total_amount?: number
  installment_amount?: number
  count: number
  category_id: number
  payment_method_id: number
  purchase_date: string
}

export interface InstallmentPlanResponse {
  id: number
  description: string
  total_amount: number
  installment_count: number
  installment_amount: number
  start_month: string
  payment_method_id: number
}

export interface InvoiceEntry {
  date: string
  title: string
  amount: number
  category_name: string
}

export interface InvoiceSummary {
  month: string
  total: number
  total_remaining: number
  entries: InvoiceEntry[]
}

export interface InstallmentCategoryOption {
  id: number
  name: string
  direction: string
  is_budget_relevant: boolean
  is_active: boolean
}
